### PR TITLE
Nisha | Test | add export button in inventory master

### PR DESCRIPTION
**Pull request by nisha-walkover**
https://app.clickup.com/t/86cvmw80z

### CODE REVIEW

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html [hunk 1]
```diff
@@ -0,0 +1,337 @@
+  <div class="modal-header">
+    <h3>Export Inventory Master</h3>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html]: Translation

File: apps/web-giddh/src/app/models/api-models/DaybookRequest.ts [hunk 2]
```diff
@@ -33,6 +33,7 @@ export class ExportBodyRequest {
     q?: any;
     tagNames?: any[];
     includeTag?: boolean;
+    groupUniqueNames?: any;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/models/api-models/DaybookRequest.ts:36]: Check this

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+    <button aria-hidden="true" class="close" mat-dialog-close>
+      ×
+    </button>
+  </div>
+  <mat-dialog-content class="overflow-hidden">
+    <div class="modal-body">
+      <p class="pd-l15 pb-3">
+        <span> {{ inputData?.commonLocaleData?.app_note }} : </span>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html]:
```suggestion
        {{ inputData?.commonLocaleData?.app_note }} :  {{ inputData?.localeData?.export_note }}
```

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+  <mat-dialog-content class="overflow-hidden">
+    <div class="modal-body">
+      <p class="pd-l15 pb-3">
+        <span> {{ inputData?.commonLocaleData?.app_note }} : </span> 
+        {{ inputData?.localeData?.export_note }}
+      </p>
+      <form [formGroup]="exportForm" class="d-flex justify-content-between px-3 pt-2 overflow-auto">
+        <div>
+          <div class="mr-b15">
+            <label class="label-width">Stock Group Name :</label>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html]: Translation

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+            <label class="p-0 ml-2 toggle-width">
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html]: why use label tag again ??? 

check and fix everywhere in this page

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+                    exportForm.get("stockGroupName")?.value
+                    ? inputData?.commonLocaleData?.app_yes
+                    : inputData?.commonLocaleData?.app_no
+                  }}
+                </span>
+              </mat-slide-toggle>
+            </label>
+          </div>
+          <div class="mr-b15">
+            <label class="label-width">Stock Name :</label>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html]: Translation

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+                    exportForm.get("stockName")?.value
+                    ? inputData?.commonLocaleData?.app_yes
+                    : inputData?.commonLocaleData?.app_no
+                  }}
+                </span>
+              </mat-slide-toggle>
+            </label>
+          </div>
+          <div class="mr-b15">
+            <label class="label-width">Unique Name :</label>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html]: Translation in over all page

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.scss [hunk 8]
```diff
@@ -0,0 +1,17 @@
+.modal-body {
+    p {
+        color: var(--color-sidebar-content-text);
+    }
+    form {
+        height: 290px;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.scss]:
```suggestion
        height: clamp(215px, 290px, 290px);
```

Optional changes

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+  ) { }
+
+    /**
+   *  This hook will use for component initialization
+   *
+   * @memberof ExportInventoryMasterComponent
+   */
+  ngOnInit() {
+    this.initExportForm();
+  }
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts:38]:
```suggestion
  }
  
```

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+    private router: Router,
+    private formBuilder: FormBuilder,
+  ) { }
+
+    /**
+   *  This hook will use for component initialization
+   *
+   * @memberof ExportInventoryMasterComponent
+   */
+  ngOnInit() {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts]:
```suggestion
  public ngOnInit(): void {
```

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+
+
+  /**
+   * Exports group/stock details
+   *
+   * @memberof ExportInventoryMasterComponent
+   */
+  public exportInventory(): void {
+    let exportRequest: ExportBodyRequest = new ExportBodyRequest();
+    console.log(exportRequest)
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts]: remove console

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+    if (formValue.customField2Heading) {
+      exportRequest.columnsToExport?.push("Custom field 2 heading");
+    }
+    if (formValue.customField2Value) {
+      exportRequest.columnsToExport?.push("Custom field 2 value");
+    }
+    if (formValue.skuCode) {
+      exportRequest.columnsToExport?.push("SKU code");
+    }
+    console.log(exportRequest)
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts]: remove console

File: apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.html [hunk 13]
```diff
@@ -118,6 +118,9 @@
                 </div>
             </div>
             <div class="inventory-btnside">
+                <div class="d-flex justify-content-end mt-1">
+                    <button mat-stroked-button color="primary" (click)="exportInventoryMaster()">Export</button>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.html]: translation

File: apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.ts [hunk 14]
```diff
@@ -53,7 +55,8 @@ export class InventoryMasterComponent implements OnInit, OnDestroy {
     constructor(
         private inventoryService: InventoryService,
         private route: ActivatedRoute,
-        private scrollDispatcher: ScrollDispatcher
+        private scrollDispatcher: ScrollDispatcher,
+        public dialog: MatDialog,
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.ts]:
```suggestion
        public dialog: MatDialog
```

File: apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.ts [hunk 15]
```diff
@@ -536,4 +539,20 @@ export class InventoryMasterComponent implements OnInit, OnDestroy {
             this.breadcrumbs = breadcrumbs;
         }
     }
+
+    // export button
+    public exportInventoryMaster() {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.ts]: Write proper doc with jsdoc

File: apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.ts [hunk 16]
```diff
@@ -536,4 +539,20 @@ export class InventoryMasterComponent implements OnInit, OnDestroy {
...
+        let exportData = {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.ts]:
```suggestion
        const exportData = {
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html (hunk 1 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html]: done

File: apps/web-giddh/src/app/models/api-models/DaybookRequest.ts (hunk 2 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/models/api-models/DaybookRequest.ts:36]: done

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html (hunk 5 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html]: done

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html (hunk 6 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html]: done

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html (hunk 7 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html]: done

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts (hunk 10 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts (hunk 12 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.html (hunk 13 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.html]: done

File: apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.ts (hunk 14 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.ts (hunk 15 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.ts (hunk 16 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.scss (hunk 8 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.scss]: done

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts (hunk 11 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html [hunk 17]
```diff
@@ -0,0 +1,295 @@
...
+                    exportForm.get("skuCode")?.value
+                    ? inputData?.commonLocaleData?.app_yes
+                    : inputData?.commonLocaleData?.app_no
+                  }}
+                </span>
+              </mat-slide-toggle>
+          </div>
+        </div>
+      </form>
+      <!-- <button mat-stroked-button color="primary" class="mt-3" (click)="exportInventory()">
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html]: Remove Comment

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html (hunk 17 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.html]: done

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts (hunk 9 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts:38]: done

**Review by dvCodeWorld — APPROVED** [review]

File: apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.ts [hunk 18]
```diff
@@ -536,4 +539,24 @@ export class InventoryMasterComponent implements OnInit, OnDestroy {
             this.breadcrumbs = breadcrumbs;
         }
     }
+
+    /**
+     * Export inventory master detail
+     *
+     * @memberof InventoryMasterComponent
+     */
+    public exportInventoryMaster() {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.ts]:
```suggestion
    public exportInventoryMaster(): void {
```

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts [hunk 19]
```diff
@@ -0,0 +1,169 @@
...
+      exportRequest.columnsToExport?.push("SKU code");
+    }
+    this.isLoading = true;
+    this.ledgerService
+      .exportData(exportRequest)
+      .pipe(takeUntil(this.destroyed$))
+      .subscribe((response) => {
+        this.isLoading = false;
+        if (response?.status === "success") {
+          this.toaster.successToast(response?.body);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts]:
```suggestion
          this.toaster.showSnackBar("success", response?.body);
```

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts [hunk 20]
```diff
@@ -0,0 +1,169 @@
...
+          this.router.navigate(["/pages/downloads"]);
+        } else {
+          this.toaster.errorToast(response?.message);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts]:
```suggestion
          this.toaster.showSnackBar("error", response?.message);
```

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts [hunk 21]
```diff
@@ -0,0 +1,169 @@
...
+import { MAT_DIALOG_DATA } from '@angular/material/dialog';
+
+@Component({
+  selector: 'app-export-inventory-master',
+  templateUrl: './export-inventory-master.component.html',
+  styleUrls: ['./export-inventory-master.component.scss']
+})
+export class ExportInventoryMasterComponent implements OnInit {
+  /* This will hold local JSON data */
+  @Input() public localeData: any = {};
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts]: useless variable. remove

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts [hunk 22]
```diff
@@ -0,0 +1,169 @@
...
+  /* This will hold common JSON data */
+  @Input() public commonLocaleData: any = {};
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts]: useless variable. remove

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.ts (hunk 18 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/inventory-master/inventory-master.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts (hunk 19 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts (hunk 20 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts (hunk 21 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts (hunk 22 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/export-inventory-master/export-inventory-master.component.ts]: done

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**